core/commands/name/name.go

```diff
@@ -151,78 +145,76 @@ Passing --verify will verify signature against provided public key.
 			return err
 		}
 
-		var entry ipns_pb.IpnsEntry
-		err = proto.Unmarshal(b.Bytes(), &entry)
+		rec, err := ipns.UnmarshalRecord(b.Bytes())
 		if err != nil {
 			return err
 		}
 
-		encoder, err := mbase.EncoderByName("base64")
-		if err != nil {
-			return err
+		result := &IpnsInspectResult{
+			Entry: IpnsInspectEntry{},
 		}
 
-		result := &IpnsInspectResult{
-			Entry: IpnsInspectEntry{
-				Value:        string(entry.Value),
-				ValidityType: entry.ValidityType,
-				Sequence:     *entry.Sequence,
-				TTL:          entry.Ttl,
-				PublicKey:    encoder.Encode(entry.PubKey),
-				SignatureV1:  encoder.Encode(entry.SignatureV1),
-				SignatureV2:  encoder.Encode(entry.SignatureV2),
-				Data:         nil,
-			},
+		// Best effort to get the fields. Show everything we can.
+		if v, err := rec.Value(); err == nil {
+			result.Entry.Value = &v
 		}
 
-		if len(entry.Data) != 0 {
-			// This is hacky. The variable node (datamodel.Node) doesn't directly marshal
-			// to JSON. Therefore, we need to first decode from DAG-CBOR, then encode in
-			// DAG-JSON and finally unmarshal it from JSON. Since DAG-JSON is a subset
-			// of JSON, that should work. Then, we can store the final value in the
-			// result.Entry.Data for further inspection.
-			node, err := ipld.Decode(entry.Data, dagcbor.Decode)
-			if err != nil {
-				return err
-			}
+		if v, err := rec.ValidityType(); err == nil {
+			result.Entry.ValidityType = &v
+		}
 
-			var buf bytes.Buffer
-			err = dagjson.Encode(node, &buf)
-			if err != nil {
-				return err
-			}
+		if v, err := rec.Validity(); err == nil {
+			result.Entry.Validity = &v
+		}
 
-			err = json.Unmarshal(buf.Bytes(), &result.Entry.Data)
-			if err != nil {
-				return err
-			}
+		if v, err := rec.Sequence(); err == nil {
+			result.Entry.Sequence = &v
 		}
 
-		validity, err := ipns.GetEOL(&entry)
-		if err == nil {
-			result.Entry.Validity = &validity
+		if v, err := rec.TTL(); err == nil {
+			result.Entry.TTL = &v
 		}
 
-		verify, ok := req.Options["verify"].(string)
-		if ok {
-			key := strings.TrimPrefix(verify, "/ipns/")
-			id, err := peer.Decode(key)
+		// Here we need the raw protobuf just to decide the version.
+		var pbRecord ipns_pb.IpnsRecord
+		err = proto.Unmarshal(b.Bytes(), &pbRecord)
+		if err != nil {
+			return err
+		}
+		if pbRecord.SignatureV1 != nil || pbRecord.Value != nil {
+			if pbRecord.Data != nil {
+				result.Version = "V1+V2"
+			} else {
+				result.Version = "V2"
+			}
```

Comment:
:thought_balloon: Is this a bug? If the Data is missing, it can't be V2. 
Seems we miss test for this.

Reply:
Yeah, this wasn;t correct at all. Fixed and added to tests.